grype/db/v6/affected_package_store.go

```diff
@@ -137,34 +211,170 @@ func (s *affectedPackageStore) getPackageByNameAndDistro(packageName string, con
 	return pkgs, nil
 }
 
-func (s *affectedPackageStore) handlePacakge(query *gorm.DB, packageName string, config GetAffectedPackageOptions) *gorm.DB {
-	query = query.Where("packages.name = ?", packageName)
-
-	if config.PackageType != "" {
-		query = query.Where("packages.type = ?", config.PackageType)
+func (s *affectedPackageStore) resolveDistro(d DistroSpecifier) ([]OperatingSystem, error) {
+	if d.Name == "" && d.Codename == "" {
+		return nil, ErrMissingDistroIdentification
 	}
-	return query
-}
 
-func (s *affectedPackageStore) handleDistro(query *gorm.DB, d *DistroSpecifier) *gorm.DB {
-	if d == AnyDistroSpecified {
-		return query
+	// search for aliases for the given distro; we intentionally map some OSs to other OSs in terms of
+	// vulnerability (e.g. `centos` is an alias for `rhel`). If an alias is found always use that alias in
+	// searches (there will never be anything in the DB for aliased distros).
```

Comment:
So what happens here when we do add affected entries for a distro that was previous an alias (like when we bring in data for `almalinux`)?  I think what would happen is that the alias would no longer exist in the grype db that is published that now contains the almalinux data, but just want to validate that assumption.

Reply:
And all prior clients would adjust correctly to it becoming a comprehensive vulnerability dataset rather than an alias to `rhel` data